modules/twinklexfd.js

```diff
@@ -203,9 +203,61 @@ Twinkle.xfd.callback.change_category = function twinklexfdCallbackChangeCategory
+			"Geographic/Asia/Middle East": ["Middle East", "Iran", "Iraq", "Israel", "Jordan", "Kuwait", "Lebanon", "Libya", "Palestine", "Saudi Arabia", "Syria", "United Arab Emirates", "Yemen", "Qatar"],
+			"Geographic/Europe": ["Europe", "Albania", "Armenia", "Austria", "Azerbaijan", "Belarus", "Belgium", "Bosnia and Herzegovina", "Bulgaria", "Croatia", "Cyprus", "Czech Republic", "Denmark", "Estonia", "Finland", "France", "Georgia (country)", "Germany", "Greece", "Hungary", "Iceland", "Ireland", "Italy", "Jersey", "Kosovo", "Latvia", "Lithuania", "Luxembourg", "Macedonia", "Malta", "Moldova", "Montenegro", "Netherlands", "Norway", "Poland", "Portugal", "Romania", "Russia", "Serbia", "Slovakia", "Slovenia", "Spain", "Sweden", "Switzerland", "Turkey", "Ukraine", "Yugoslavia"],
+			"Geographic/Europe/United Kingdom": ["United Kingdom", "England", "Northern Ireland", "Scotland", "Wales"],
+			"Geographic/Oceania": ["Oceania", "Antarctica", "Australia", "New Zealand"],
+			"Geographic/Americas/Canada": ["Canada", "British Columbia", "Manitoba", "Nova Scotia", "Ontario", "Quebec", "Alberta"],
+			"Geographic/Americas/Latin America": ["Latin America", "Caribbean", "South America", "Argentina", "Barbados", "Belize", "Bolivia", "Brazil", "Chile", "Colombia", "Cuba", "Ecuador", "El Salvador", "Guatemala", "Haiti", "Mexico", "Nicaragua", "Panama", "Paraguay", "Peru", "Puerto Rico", "Trinidad and Tobago", "Uruguay", "Venezuela", "Grenada"],
+			"Geographic/Americas/USA": ["United States of America", "Alabama", "Alaska", "Arizona", "Arkansas", "California", "Colorado", "Connecticut", "Delaware", "Florida", "Georgia (U.S. state)", "Hawaii", "Idaho", "Illinois", "Indiana", "Iowa", "Kansas", "Kentucky", "Louisiana", "Maine", "Maryland", "Massachusetts", "Michigan", "Minnesota", "Mississippi", "Missouri", "Montana", "Nebraska", "Nevada", "New Hampshire", "New Jersey", "New Mexico", "New York", "North Carolina", "North Dakota", "Ohio", "Oklahoma", "Oregon", "Pennsylvania", "Rhode Island", "South Carolina", "South Dakota", "Tennessee", "Texas", "Utah", "Vermont", "Virginia", "Washington", "Washington, D.C.", "West Virginia", "Wisconsin", "Wyoming"],
+			"Geographic/Unsorted": ["Islands"]
+		};
+
+		var delsort = work_area.append( {
+			type: 'select',
+			multiple: true,
+			name: 'delsort',
+			label: 'Choose deletion sorting pages to list in: '
+		} );
+
+		$.each(delsortCategories, function(grpname, list) {
+			var grp = delsort.append( { type: 'optgroup', label: grpname } );
+			list.forEach(function(item) {
+				grp.append( { type: 'option', label: item, value: item } );
+			});
+		});
+
 		appendReasonBox();
 		work_area = work_area.render();
 		old_area.parentNode.replaceChild( work_area, old_area );
+		mw.loader.using('jquery.chosen').then(function chosenInit() {
+			$(work_area).find('[name=delsort]')
+				.attr('data-placeholder', 'Select delsort pages')
+				.chosen({width: "100%"});
```

Comment:
CSS isn't my strong suit but I don't suppose there's any way to get this to start on the same line as the label, but then be at 100% when multiline?  I think it's either/or.

Reply:
Don't really follow. Are u saying that the input box should be on the same line as the label? I don't think so since the label is kinda long, also if there are multiple delsorts, they'd take up space.

Reply:
I don't think it's possible, but what I *really* want is for it to be on the same line, but then when it would overflow, the input would be on it's own line.  Not something I intended for you to fix, just wishin'

Reply:
Not possible it seems. But even if it was, I'd say such "jumping" elements are poor user interfaces.